pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/channel/ServiceUnitStateChannelImpl.java

```diff
@@ -503,13 +503,22 @@ public CompletableFuture<Optional<String>> getOwnerAsync(String serviceUnit) {
             case Splitting -> {
                 return CompletableFuture.completedFuture(Optional.of(data.sourceBroker()));
             }
-            case Assigning, Releasing -> {
+            case Assigning -> {
                 return deferGetOwnerRequest(serviceUnit).whenComplete((__, e) -> {
                     if (e != null) {
                         ownerLookUpCounters.get(state).getFailure().incrementAndGet();
                     }
-                }).thenApply(
-                        broker -> broker == null ? Optional.empty() : Optional.of(broker));
+                }).thenApply(broker -> broker == null ? Optional.empty() : Optional.of(broker));
+            }
+            case Releasing -> {
+                if (isTransferCommand(data)) {
+                    return deferGetOwnerRequest(serviceUnit).whenComplete((__, e) -> {
+                        if (e != null) {
+                            ownerLookUpCounters.get(state).getFailure().incrementAndGet();
+                        }
+                    }).thenApply(broker -> broker == null ? Optional.empty() : Optional.of(broker));
+                }
+                return CompletableFuture.completedFuture(Optional.of(data.sourceBroker()));
```

Comment:
@heesung-sn It should return the current owner when it is in the `Releasing` state, WDYT?

Reply:
I am worried that this might cause some racing conditions (e.g. new pub/subs are created after closeServiceUnit..)

Reply:
I rechecked the `OwnershipCache` logic, and it will set the bundle state to inactive, so we should return `empty` here. I changed back.